src/transformers/models/vits/tokenization_vits.py

```diff
@@ -0,0 +1,168 @@
+from ...utils import is_phonemizer_available, logging
+
+
+if is_phonemizer_available():
+    import phonemizer
+
+
+logger = logging.get_logger(__name__)
+
+VOCAB_FILES_NAMES = {"vocab_file": "vocab.json"}
+
+PRETRAINED_VOCAB_FILES_MAP = {
+    "vocab_file": {
+        "Matthijs/mms-tts-eng": "https://huggingface.co/Matthijs/mms-tts-eng/resolve/main/vocab.json",
+    }
+}
+
+PRETRAINED_POSITIONAL_EMBEDDINGS_SIZES = {
+    # This model does not have a maximum input length.
+    "Matthijs/mms-tts-eng": 4096,
+}
+
+
+class VitsTokenizer(PreTrainedTokenizer):
+    """
+    Construct a VITS tokenizer. Also supports MMS-TTS.
+
+    This tokenizer inherits from [`PreTrainedTokenizer`] which contains most of the main methods. Users should refer to
+    this superclass for more information regarding those methods.
+
+    Args:
+        vocab_file (`str`):
+            Path to the vocabulary file.
+        language (`str`, *optional*):
+            Language identifier.
+        add_blank (`bool`, *optional*, defaults to `True`):
+            Whether to insert token id 0 in between the other tokens.
+        phonemize (`bool`, *optional*, defaults to `True`):
+            Whether to convert the input text into phonemes.
+    """
+
+    vocab_files_names = VOCAB_FILES_NAMES
+    pretrained_vocab_files_map = PRETRAINED_VOCAB_FILES_MAP
+    max_model_input_sizes = PRETRAINED_POSITIONAL_EMBEDDINGS_SIZES
+    model_input_names = ["input_ids", "attention_mask"]
+
+    def __init__(
+        self,
+        vocab_file,
+        pad_token="<pad>",
+        unk_token="<unk>",
+        language=None,
+        add_blank=True,
+        phonemize=True,
+        **kwargs,
+    ) -> None:
+        super().__init__(
+            pad_token=pad_token,
+            unk_token=unk_token,
+            language=language,
+            add_blank=add_blank,
+            phonemize=phonemize,
+            **kwargs,
+        )
+
+        with open(vocab_file, encoding="utf-8") as vocab_handle:
+            self.encoder = json.load(vocab_handle)
+
+        # The original model pads with token_id 0, but that doesn't work here
+        # as token_id 0 is an actual character. So we add a fake padding token
+        # and then filter that out in the model.
+        if pad_token not in self.encoder:
+            self.encoder[pad_token] = len(self.encoder)
+        if unk_token not in self.encoder:
+            self.encoder[unk_token] = len(self.encoder)
+
+        self.decoder = {v: k for k, v in self.encoder.items()}
+        self.language = language
+        self.add_blank = add_blank
+        self.phonemize = phonemize
+
+    @property
+    def vocab_size(self):
+        return len(self.encoder)
+
+    def get_vocab(self):
+        vocab = {self.convert_ids_to_tokens(i): i for i in range(self.vocab_size)}
+        return vocab
+
+    def _tokenize(self, text: str) -> List[str]:
+        """Tokenize a string."""
+        text = self._preprocess_char(text.lower())
```

Comment:
This should go in the `tokenize` function, otherwise it is applied on each split (if you properly use the special tokens, text is split on special tokens)

Reply:
Resolved in https://github.com/huggingface/transformers/pull/24085/commits/89254691ec71641c6ebdc69a9e6695317bf09fee